clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1497,4 +1495,13 @@ public void handleResponse(AbstractResponse response) {
             }
         }
     }
+
+    private boolean isFatalException(Errors error) {
+        return error == Errors.TRANSACTIONAL_ID_AUTHORIZATION_FAILED
+                || error == Errors.INVALID_PRODUCER_EPOCH
+                || error == Errors.UNSUPPORTED_FOR_MESSAGE_FORMAT
+                || error == Errors.FENCED_INSTANCE_ID
+                || error == Errors.UNKNOWN_MEMBER_ID
```

Comment:
If we make these fatal errors, then the user will have to close the producer and instantiate a new one. I think as long as the producer can still abort the transaction, they should be abortable errors.

Reply:
The purpose of throwing these exceptions should be catching user's attention IMO, as this indicates something going wrong on the consumer side. So if this is abortable error, how could we inform application(i.e consumer) level that sth is going wrong? Or we believe that as long as the consumer is in a bad state, it will get itself fenced eventually?

Reply:
Abortable errors are still propagated to the user. But the user can still abort the transaction and continue with the same producer.